src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -260,38 +260,101 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
             newRd.Day_of_Month__c != oldRd.Day_of_Month__c ||
             newRd.Status__c != oldRd.Status__c);
     }
+
     /***
     * @description Handles setting next donation date for updated recurring donations
```

Comment:
Just so that I'm sure, the intent of this method is the following:

- Perform a full analysis and rebuild of the Schedule records for a Recurring Donation
- Do not commit the schedule records to the database
- Use the active schedules for the RD to calculate the Next Donation Date and store onto the RD record (in a Before Update context, thus avoiding the extra DML against the object).


If correct, I suggest updating the method comment to provide a bit more background on the why for this method.

Reply:
Though if that is accurate, shouldn't this logic be roughly the same as the `synchronizeSchedules` method, maybe less the DML operations?

Reply:
I've refactored the code so that the commonalities between the two methods are abstracted out into separate methods that can be called from each - getExistingSchedules() and getRecurringDonationsNeedingScheduleUpdates().  There is a key difference, though.  The before update version unconditionally sets next donation date, so it needs to look at all updates, not just those that qualify for schedule updates.

Reply:
It _may_ be true that the next donation date can't ever change unless there is an update to the RD.  If so, it would be true that the methods should be identical.  I'm very reluctant to make that assumption, though.  If it isn't 100% true both now and in the future, that would be a very difficult issue to track down.